src/navigator-card-body.jsx

```diff
@@ -309,7 +310,72 @@
         }
     };
 
-    const handleClick = (ev, file) => {
+    const getNameForEvent = event => {
+        let name = null;
+
+        // We can't stick a data-item on the <label> of the cardHeader's
+        // selectableActions so we use the for attribute instead.
+        const for_attr = event.target.getAttribute("for");
+
+        // HACK: The current selected implementation, always sets the selected
+        // to the current directory. This is for the case where a user clicks
+        // on the sidebar menu or when they use the context menu to click on the
+        // "background layer". This is unneeded complexity and the caller
+        // should instead fallback to the current directory when nothing is
+        // selected.
+        if (for_attr !== null) {
+            const match = for_attr.match(/card-item-(.*?)-selectable-action/);
+            if (match === null || match.length === 2) {
+                name = match[1];
+            }
+        }
+
+        const file = sortedFiles?.find(file => file.name === name);
+        if (!file)
+            return null;
+
+        return file;
+    };
+
+    const getFileForEvent = event => {
+        let data_item = null;
+        const max_depth = 10;
+        let elem = event.target;
+        for (let i = 0; i < max_depth; i++) {
+            data_item = elem.getAttribute("data-item");
+            if (data_item)
+                break;
+
+            if (elem.parentElement)
+                elem = elem.parentElement;
+            else
+                break;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-249-20240207-125527-92469164-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)